extensions/2.0/Vendor/NV_materials_mdl/README.md

```diff
@@ -0,0 +1,511 @@
+
+MDL materials are defined through functions in the language that return the built-in type `material`. This extension operates on function calls of functions defined in referenced MDL modules. These function calls are defined in the `functionCalls` array property which is defined in the top-level property `extensions.NV_materials_mdl`. Modules which are referenced by the function calls and types are defined in the `modules` array. Lastly, BSDF measurement resources are defined in the `bsdfMeasurements` array.
+
+```json
+"extensions": {
+    "NV_materials_mdl": {
+        "modules": [
+            ...
+        ],
+        "bsdfMeasurements": [
+            ...
+        ],
+        "functionCalls": [
+            ...
+        ]
+    }
+}
+```
+
+## Adding a MDL Material to an Existing Material
+MDL materials must be referenced in the material by specifying an index into the `functionCalls` array using the `funcionCall` property within a material's `extensions.NV_materials_mdl` property. This function call index represents the root node of a graph of function calls and must have the built-in return type `material`.
+
+```json
+"materials": [
+    {
+        "extensions": {
+            "NV_materials_mdl": {
+                "functionCall": 1
+            }
+        }
+    }
+]
+```
+
+## Defining MDL Modules
+All modules that are referenced by the function calls in the `functionCalls` array must be listed in the `modules` array. A `module` has the following properties:
+
+| Property | Type | Description | Required |
+|:---------|:-----|:------------|:---------|
+| `uri` | `string` | The URI (or IRI) of the MDL module. | No |
+| `bufferView` | `integer` | The index of the bufferView that contains the MDL module. | No |
+| `mimeType` | `string` | The media type of the bufferView's data. | No |
+| `modulePath` | `string` | Relative path of the module. | No |
+| `name` | `string` | glTF name of the module. | No, Default: `""` |
+
+Each module defines either the `uri` property, which contains the URI (or IRI) of an external MDL module file or a data-URI with embedded data, or the `bufferView` property, which points to a bufferView object that contains an MDL module. If the `bufferView` property is used then `mimeType` must also be defined and have the value `application/vnd.mdl`. If a data-URI or bufferView object is used then the `modulePath` property must be defined, which specifies a path relative to the glTF file that is used for importing by other modules, e.g. if an MDL module that has the path `./my/embedded/module.mdl` relative to the glTF file is embedded into a bufferView object then the `modulePath` property must have this path as its value.
```

Comment:
- nit: Missing comma after 'e.g.'
- Mention that data-URI must have the `application/vnd.mdl` media type.
- Is it allowed to have both `uri` and `modulePath` defined at the same time?

Reply:
Yes, `uri` and `modulePath` must be defined at the same if `uri` contains a data-URI, since `modulePath` acts as a "virtual file path" for the cases where no real file path is provided.

Reply:
Besides data-URI, is it allowed to have `modulePath` and `uri` simultaneously?

Reply:
If the URI isn't a data-URI, then they shouldn't be allowed simultaneously. Or in other words, the implementation should ignore `modulePath` then.

Reply:
I clarified this in the README and schema. Now it should be clear that `modulePath` must not be defined if `uri` is defined and isn't a data-URI.